python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1303 @@
+            "DETECTOR": "One kernel per detector",
+        }
+    )
+
+
+class MakeBrighterFatterKernelTaskRunner(pipeBase.TaskRunner):
+    """Subclass of TaskRunner for the MakeBrighterFatterKernelTask.
+
+    This transforms the processed arguments generated by the ArgumentParser into the arguments expected by
+    makeBrighterFatterKernelTask.run().
+
+    makeBrighterFatterKernelTask.run() takes a two arguments, one of which is the dataRef (as usual),
+    and the other is the list of visitPairs, in the form of a list of tuples.
+    This list is supplied on the command line as documented, and this class parses that,
+    and passes the parsed version to the run() method.
+
+    See pipeBase.TaskRunner for more information.
+    """
+
+    @staticmethod
+    def getTargetList(parsedCmd, **kwargs):
+        """Parse the visit list and pass through explicitly."""
+        visitPairs = [(int(visit.split(",")[0]), int(visit.split(",")[1])) for visit in parsedCmd.visitPairs]
+        return pipeBase.TaskRunner.getTargetList(parsedCmd, visitPairs=visitPairs, **kwargs)
+
+
+class BrighterFatterKernelTaskDataIdContainer(pipeBase.DataIdContainer):
+    """A DataIdContainer for the MakeBrighterFatterKernelTask."""
+
+    def makeDataRefList(self, namespace):
+        """Compute refList based on idList.
+
+        This method must be defined as the dataset does not exist before this task is run.
+
+        Parameters
+        ----------
+        namespace
+            Results of parsing command-line (with ``butler`` and ``log`` elements).
+
+        Notes
+        -----
+        Not called if ``add_id_argument`` called with ``doMakeDataRefList=False``.
+        Note that this is almost a copy-and-paste of the vanilla implementation, but without checking
+        if the datasets already exist as this task exists to make them.
+        """
+        if self.datasetType is None:
+            raise RuntimeError("Must call setDatasetType first")
+        butler = namespace.butler
+        for dataId in self.idList:
+            refList = list(butler.subset(datasetType=self.datasetType, level=self.level, dataId=dataId))
+            # exclude nonexistent data
+            # this is a recursive test, e.g. for the sake of "raw" data
+            if not refList:
+                namespace.log.warn("No data found for dataId=%s", dataId)
+                continue
+            self.refList += refList
+
+
+class MakeBrighterFatterKernelTask(pipeBase.CmdLineTask):
+    """Brighter-fatter effect correction-kernel calculation task.
+
+    A command line task for calculating the brighter-fatter correction
+    kernel from pairs of flat-field images (with the same exposure length).
+
+    The following operations are performed:
+
+    - The configurable isr task is called, which unpersists and assembles the
+      raw images, and performs the selected instrument signature removal tasks.
+      The appropriacy of the selected isr components is checked.
```

Comment:
"appropriacy" is an great word (one new to me), but the definitions I found don't work here. Perhaps it would be more helpful to say that certain ISR steps must be performed. If they can be briefly explained then do it here. Then point to the config fields that specify the checks.

Reply:
OK, will fix. But for reference, the Google definition I get is:
```
appropriacy
əˈprəʊprɪəsi/
noun
noun: appropriacy; plural noun: appropriacies

    the extent to which something is suitable or proper in the circumstances.
    "the appropriacy of the methodology employed"
```
which is the way in which I intended it, and seems to fit well, to my ears at least 🙂

Reply:
I'm not sure what you mean by "the config fields that specify the checks" though, as there are none - the isr config is always checked. I guess you mean "the config fields that are specified _by_ the checks", _i.e._ which fields are checked, right? If so, I'll mention the most important and most-likely-misconfigured ones, as the list is quite long, and the code can define the rest (they live in three very self-explanatory lists after all).